### PR TITLE
Move memory and parallelism settings for performance test projects to gradle.properties

### PR DESCRIPTION
**Pull request by wolfs**
So anybody running a performance test project will use the same settings as our performance tests.

### CODE REVIEW

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle [hunk 1]
```diff
@@ -176,24 +190,28 @@ performanceTest.registerTestProject("nativeMonolithicOverlapping", MonolithicNat
     // number of source files to generate in each project of each type
     // this generates 200 C, 200 C++, 200 headers and 400 unused C/C++ files
     sourceFiles = 200
+    daemonMemory = '2500m'
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle:193]: Nit: shall we use multiples of 1024, or something like `2.5G`?

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild/performance/generator/tasks/NativeProjectWithDepsGeneratorTask.groovy [hunk 2]
```diff
@@ -136,6 +140,21 @@ class NativeProjectWithDepsGeneratorTask extends TemplateProjectGeneratorTask {
...
+                copySpec.from(new File(resolveTemplate("gradle-properties"), "gradle.properties"))
+                copySpec.into(destDir)
+                copySpec.expand([
+                    daemonMemory: daemonMemory,
+                    maxWorkers: 4,
+                    parallel: true
+                ])
+            }
+        })
+    }
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild/performance/generator/tasks/NativeProjectWithDepsGeneratorTask.groovy:156]: Nit: can we use a closure here?

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AndroidTestProject.groovy [hunk 3]
```diff
@@ -56,14 +53,10 @@ class AndroidTestProject implements TestProject {
...
     void configure(CrossVersionPerformanceTestRunner runner) {
-        runner.gradleOpts = ["-Xms$memory", "-Xmx$memory"]
     }
 
     @Override
     void configure(GradleBuildExperimentSpec.GradleBuilder builder) {
-        builder.invocation {
-            gradleOpts("-Xms$memory", "-Xmx$memory")
-        }
     }
```
Comment (lptr) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AndroidTestProject.groovy:60]: Do we still need these methods?

**Review by lptr — COMMENTED** [review]
In general LGTM, left a few comments.

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle [hunk 4]
```diff
@@ -52,18 +52,20 @@ performanceTest.registerTestProject("withVerboseTestNG", JvmProjectGeneratorTask
     projects = 1
     sourceFiles = 10
     subProjectTemplates << 'with-verbose-testng'
+    daemonMemory = '256m'
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle:55]: Would it make sense to extract these to constants, like `SMALL_MEMORY`, `LARGE_MEMORY` etc.?

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle (hunk 1 above)
Comment (wolfs) [reply buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle:193]: `2.5G` is not a valid `Xmx` option. I switched to `2560`

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AndroidTestProject.groovy (hunk 3 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AndroidTestProject.groovy:60]: The AGP test projects add a Gradle property to enable the IDE mode via this method: `-Pandroid.injected.invoked.from.ide=true`. Maybe we should add this property to the `gradle.properties` file of those projects as well. WDYT?

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle (hunk 4 above)
Comment (wolfs) [reply buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle:55]: I don't think that makes sense. The memory requirements would not necessarily change for all projects which use the same constant at the same time. The memory is a per project setting and accidentally some project use the same value.

File: buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild/performance/generator/tasks/NativeProjectWithDepsGeneratorTask.groovy (hunk 2 above)
Comment (wolfs) [reply buildSrc/subprojects/performance-testing/src/main/groovy/gradlebuild/performance/generator/tasks/NativeProjectWithDepsGeneratorTask.groovy:150]: Yes, done.

**Comment by wolfs** [comment]
@lptr If you have more comments, I can address them in a follow up PR.

**Pull request merged**